src/audio/tdfb/tdfb.c

```diff
@@ -358,6 +523,49 @@ static int tdfb_cmd_set_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int tdfb_cmd_enum_set(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	int ret = 0;
+
+	switch (cdata->index) {
+	case CTRL_INDEX_DIRECTION:
+		if (cdata->num_elems) {
+			cd->az_value = cdata->chanv[0].value;
+			cd->update = true;
+		}
+		break;
+	case CTRL_INDEX_ONOFF:
```

Comment:
we should really enable the proper switch control for this

Reply:
Ah, sorry, I remember we talked about it but I forgot. I'll update this PR. Thanks!

Reply:
This PR version is updated with an on/off switch.